lib/server.js

```diff
@@ -32,6 +32,10 @@ var Router = util.safeRequire('react-router');
 var match = Router.match;
 var RoutingContext = Router.RoutingContext;
 
+//for creating a URL
+var createMemoryHistory = require('history').createMemoryHistory;
+var createLocation = createMemoryHistory().createLocation;
```

Comment:
actually its not really important.

i was solving the need to having req.url which is passed as location in `match` function

```js
 match({ routes:createOptions.routes,
          location:createLocation(thing)},
```

```js
const props = {
          history,
          location,
          params,
          route,
          routeParams,
          routes
        }
```
please suggest an alternative since req.url is not exposed here.